controllers/cassandradatacenter_controller_test.go

```diff
@@ -0,0 +1,237 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	cassdcapi "github.com/datastax/cass-operator/operator/pkg/apis/cassandra/v1beta1"
+	api "github.com/k8ssandra/reaper-operator/api/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	CassandraDatacenterName = "dc1"
+	ControllerTestNamespace = "dc-test"
+)
+
+var _ = Describe("CassandraDatacenter controller testing", func() {
+	BeforeEach(func() {
+		// Add any setup steps that needs to be executed before each test
+	})
+
+	AfterEach(func() {
+		// Add any teardown steps that needs to be executed after each test
+	})
+
+	It("checking Reaper cluster status", func() {
+		By("expecting CassandraDatacenter creation")
+		testNamespace := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ControllerTestNamespace,
+			},
+		}
+		Expect(k8sClient.Create(context.Background(), testNamespace)).Should(Succeed())
+		reaper := createReaper(ControllerTestNamespace)
+		Expect(k8sClient.Create(context.Background(), reaper)).Should(Succeed())
+
+		cassdcKey := types.NamespacedName{
+			Name:      CassandraDatacenterName,
+			Namespace: ControllerTestNamespace,
+		}
+
+		testDc := &cassdcapi.CassandraDatacenter{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        cassdcKey.Name,
+				Namespace:   cassdcKey.Namespace,
+				Annotations: map[string]string{},
+			},
+			Spec: cassdcapi.CassandraDatacenterSpec{
+				ClusterName:   "test-dc",
+				ServerType:    "cassandra",
+				ServerVersion: "3.11.7",
+				Size:          3,
+			},
+			Status: cassdcapi.CassandraDatacenterStatus{
+				CassandraOperatorProgress: cassdcapi.ProgressReady,
+				Conditions: []cassdcapi.DatacenterCondition{
+					{
+						Status: corev1.ConditionTrue,
+						Type:   cassdcapi.DatacenterReady,
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(context.Background(), testDc)).Should(Succeed())
```

Comment:
Note that this `Create()` call will not update  `testDc.Status`. You need to use either `k8sClient.Status.Update()` or `k8sClient.Status.Patch()`. The test actually passes with `testDc.Status` commented out.

I think you can leave the status out. `cassandradatacenter_controller.go` does not actually check the CassandraDatacenter status. I was debating whether we should check the status before calling `r.ReaperManager.AddClusterToReaper`. I do not think so. I think it just adds an extra, unnecessary step.

Reply:
Yeah, it does not make any difference here as that Status is not compared in the ``cassandradatacenter_controller.go``, it was just copied over from the other test originally.